src/core/linter-rules/tests-exist.js

```diff
@@ -0,0 +1,106 @@
+// @ts-check
+/**
+ * Linter rule "tests-exist".
+ * Warns about nonexistent web platform tests.
+ */
+import LinterRule from "../LinterRule.js";
+import { lang as defaultLang } from "../l10n.js";
+import { pub } from "../pubsubhub.js";
+
+const name = "tests-exist";
+
+const meta = {
+  en: {
+    description: "Non-existent Web Platform Tests",
+    howToFix: "Please fix the tests mentioned.",
+    help: "See developer console.",
+  },
+};
+
+const lang = defaultLang in meta ? defaultLang : "en";
+
+/**
+ * Runs linter rule.
+ * @param {Object} conf The ReSpec config.
+ * @param  {Document} doc The document to be checked.
+ * @return {Promise<import("../../core/LinterRule").LinterResult>}
+ */
+async function linterFunction(conf, doc) {
+  const filesInWPT = await getFilesInWPT(conf.testSuiteURI, conf.githubAPI);
+  if (!filesInWPT) {
+    return;
+  }
+
+  const offendingElements = [];
+  const offendingTests = new Set();
+
+  /** @type {NodeListOf<HTMLElement>} */
+  const elems = doc.querySelectorAll("[data-tests]");
+  const testables = [...elems].filter(elem => elem.dataset.tests);
+
+  for (const elem of testables) {
+    const tests = elem.dataset.tests
+      .split(/,/gm)
+      .map(test => test.trim().split("#")[0])
+      .filter(test => test);
+
+    const missingTests = tests.filter(test => !filesInWPT.has(test));
+    if (missingTests.length) {
+      offendingElements.push(elem);
+      missingTests.forEach(test => offendingTests.add(test));
+    }
+  }
+
+  if (!offendingElements.length) {
+    return;
+  }
+
+  const missingTests = [...offendingTests].map(test => `\`${test}\``);
+  return {
+    name,
+    offendingElements,
+    occurrences: offendingElements.length,
+    ...meta[lang],
+    description: `${meta[lang].description}: ${missingTests.join(", ")}.`,
+  };
+}
+
+export const rule = new LinterRule(name, linterFunction);
+
+/**
+ * @param {string} testSuiteURI
+ * @param {string} githubAPIBase
+ */
+async function getFilesInWPT(testSuiteURI, githubAPIBase) {
+  let wptDirectory;
+  try {
+    const testSuiteURL = new URL(testSuiteURI);
+    if (testSuiteURL.pathname.startsWith("/web-platform-tests/wpt/")) {
+      const re = /web-platform-tests\/wpt\/(.+)/;
+      wptDirectory = testSuiteURL.pathname.match(re)[1].replace(/\//g, "");
+    } else {
+      wptDirectory = testSuiteURL.pathname.replace(/\//g, "");
+    }
+  } catch {
```

Comment:
I think we should still console.error out here... it could be a bad URL or something else in the future. This would just be mostly for us.

Reply:
Done.